From ec40793b25b840511e4dc6e738bbf032bc13833d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 17 Oct 2024 10:14:26 +0200
Subject: [PATCH] Add icons to GResource

Before that the icons were installed twice:
- once into /usr/share/icons/hicolor/scalable/status
- and once again into /usr/share/valuta/valuta/assets/icons

Add into the GResource instead, which is the usual place for these assets.
---
 src/assets/icons/meson.build | 13 -------------
 src/meson.build              |  6 ------
 src/valuta.gresource.xml     |  8 ++++++++
 3 files changed, 8 insertions(+), 19 deletions(-)
 delete mode 100644 src/assets/icons/meson.build

diff --git a/src/assets/icons/meson.build b/src/assets/icons/meson.build
deleted file mode 100644
index fb8c9a3..0000000
--- a/src/assets/icons/meson.build
+++ /dev/null
@@ -1,13 +0,0 @@
-install_data(
-  [
-    'arrow1-down-symbolic.svg',
-    'arrow1-up-symbolic.svg',
-    'dark-mode-symbolic.svg',
-    'vertical-arrows-symbolic.svg',
-    'sonar-symbolic.svg',
-    'refresh-large-symbolic.svg',
-    'code-symbolic.svg',
-    'help-about-symbolic.svg',
-  ],
-  install_dir: datadir / 'icons/hicolor/scalable/status',
-)
diff --git a/src/meson.build b/src/meson.build
index daef034..5b259c6 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -66,11 +66,5 @@ install_subdir('components', install_dir: moduledir)
 # ===== Install page sources =====
 install_subdir('pages', install_dir: moduledir)
 
-# ===== Install assets sources =====
-install_subdir('assets', install_dir: moduledir)
-
 # ===== Install provider =====
 subdir('search_provider')
-
-# ===== Install icons =====
-subdir('assets/icons')
diff --git a/src/valuta.gresource.xml b/src/valuta.gresource.xml
index 24c10e1..3fec0d4 100644
--- a/src/valuta.gresource.xml
+++ b/src/valuta.gresource.xml
@@ -1,6 +1,14 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <gresources>
   <gresource prefix="/io/github/idevecore/Valuta">
+    <file alias="icons/scalable/status/arrow1-down-symbolic.svg" preprocess="xml-stripblanks">assets/icons/arrow1-down-symbolic.svg</file>
+    <file alias="icons/scalable/status/arrow1-up-symbolic.svg" preprocess="xml-stripblanks">assets/icons/arrow1-up-symbolic.svg</file>
+    <file alias="icons/scalable/status/code-symbolic.svg" preprocess="xml-stripblanks">assets/icons/code-symbolic.svg</file>
+    <file alias="icons/scalable/status/dark-mode-symbolic.svg" preprocess="xml-stripblanks">assets/icons/dark-mode-symbolic.svg</file>
+    <file alias="icons/scalable/status/help-about-symbolic.svg" preprocess="xml-stripblanks">assets/icons/help-about-symbolic.svg</file>
+    <file alias="icons/scalable/status/refresh-large-symbolic.svg" preprocess="xml-stripblanks">assets/icons/refresh-large-symbolic.svg</file>
+    <file alias="icons/scalable/status/sonar-symbolic.svg" preprocess="xml-stripblanks">assets/icons/sonar-symbolic.svg</file>
+    <file alias="icons/scalable/status/vertical-arrows-symbolic.svg" preprocess="xml-stripblanks">assets/icons/vertical-arrows-symbolic.svg</file>
     <file compressed="true" alias="style.css">assets/style.css</file>
     <file preprocess="xml-stripblanks">window.ui</file>
     <file preprocess="xml-stripblanks">components/currency_selector/index.ui</file>
